sky/data/data_utils.py

```diff
@@ -225,3 +329,120 @@ def run_upload_cli(command: str, access_denied_message: str, bucket_name: str):
                 raise exceptions.StorageUploadError(
                     f'Upload to bucket failed for store {bucket_name}. '
                     'Please check the logs.')
+
+
+def store_rclone_config(bucket_name: str,
+                        region: str,
+                        apply_changes: bool = True):
+    """creates a configuration files for rclone - used for
+    bucket syncing and mounting """
+
+    rclone_config_path = os.path.expanduser(RCLONE_CONFIG_PATH)
+    access_key_id, secret_access_key = ibm.get_hmac_keys()
+
+    config_data = textwrap.dedent(f"""\
+        [{bucket_name}]
+        type = s3
+        provider = IBMCOS
+        access_key_id = {access_key_id}
+        secret_access_key = {secret_access_key}
+        region = {region}
+        endpoint = s3.{region}.cloud-object-storage.appdomain.cloud
+        location_constraint = {region}-smart
+        acl = private
+        """)
+
+    # create ~/.config/rclone/ if doesn't exist
+    os.makedirs(os.path.dirname(rclone_config_path), exist_ok=True)
+    # create rclone.conf if doesn't exist
+    if not os.path.isfile(rclone_config_path):
+        open('file', 'w').close()
+
+    with FileLock(rclone_config_path + '.lock'):
+        profiles_to_keep = _remove_bucket_profile_rclone(bucket_name)
+
+        # write back file without profile: [bucket_name]
+        # to which the new bucket profile is appended
+        if apply_changes:
+            with open(f'{rclone_config_path}', 'w') as file:
+                if profiles_to_keep:
+                    file.writelines(profiles_to_keep)
+                    if profiles_to_keep[-1].strip():
+                        # add a new line to config_data
+                        # if last file line contains data
+                        config_data += '\n'
+                file.write(config_data)
```

Comment:
- Neither mount nor sync should have `apply_changes` set to True, i.e. `mount_command` currently creates an excess write (which doesn't harm functionally, it's just redundant). It will be set to `False` next commit. 
- The only instance where `store_rclone_config` should be provided with `apply_changes=True`, is in `_get_bucket`, i.e. only once when creating the storage instance. 
- Overwriting is in my opinion the way to go, since I always want to update the profile with the latest data given, e.g. region and instance-id buckets are associated with. It shouldn't collide with existing profiles created externally by the user, as profiles created by sky will have a distinctive prefix, as suggested in the next conversation you created. I suggest: `Sky-{cloud_provider}`.